Migrate GLPK to be an external project (#10880)
cvc5/cvc5: INSTALL.rst

This can be used to speed up certain classes of problems for the arithmetic
implementation in cvc5. (This is not recommended for most users.)

> cvc5 is no longer compatible with the main GLPK library.

Reviewer: ```suggestion
cvc5 is not compatible with the official version of the GLPK library.
```
As far as I know it was never compatible with the official version to begin with.

To use the patched version of it, we recommend downloading it using
the ``--auto-download`` configuration flag, which applies
the patch automatically.

> Configure cvc5 with ``configure.sh --glpk`` to build with this dependency.

Reviewer: This requires the `--gpl` flag, right? Maybe add a comment that GLPK is GPL and requires this additional flag.
